pkg/apis/pipeline/v1/taskrun_types.go

```diff
@@ -82,6 +82,10 @@ type TaskRunSpec struct {
 	ComputeResources *corev1.ResourceRequirements `json:"computeResources,omitempty"`
 }
 
+type Context struct {
+	Params Params `json:"params,omitempty"`
+}
+
```

Comment:
Is there any particular reason why we want to use Param type here? I am thinking if `any/interface{}` would be a better choice here because of the wide variety of possible context types from different builders.

Reply:
I originally thought this should be unstructured (in the status), but since we'll be using these values to substitute for parameters in pipelines and tasks, it makes the most sense to use a structured type. If platforms need more flexibility, we can implement support for nested array/object params, or add more fields to "context" as needed. I'm not sure we have a use case right now for platforms to be able to add arbitrary unstructured context to a pipelinerun.